pkg/geo/wkt/lex.go

```diff
@@ -51,13 +51,43 @@ func (e *ParseError) Error() string {
 // Constant expected by parser when lexer reaches EOF.
 const eof = 0
 
+// Stack object used for parsing the geometry type.
+// layout      is the currently parsed geometry type.
```

Comment:
nit: put these comments above the functions
```type layoutStackObj struct {
// layout is the currently parsed geometry type.
	layout      geom.Layout
// baseType is a bool where true means the current context is the top-level or a base type GEOMETRYCOLLECTION.
	baseType    bool
// mustBeEmpty is a bool used to handle the edge case where a base type geometry is allowed in a GEOMETRYCOLLECTIONM
//             but only if it is EMPTY
	mustBeEmpty bool
```

i'd rename `baseType` to be `isBaseType`.

maybe `mustBeEmpty` should be called `mustBeEmptyInGeometryCollectionM`, given it's hyper-specialised use case

Reply:
I renamed `baseType` to `inBaseTypeCollection` and `mustBeEmpty` to `nextPointMustBeEmpty`